pytorch_lightning/trainer/data_loading.py

```diff
@@ -212,6 +241,19 @@ def __init__(self, num_features, dataset, *args, **kwargs):
 
         if not contains_dataset:
             dl_args.pop('dataset')
+        else:
+            # wrap the ``IterableDataset`` into a ``CaptureIterableDataset`` to record sampler states.
+            if _fault_tolerant_enabled() and isinstance(dl_args["dataset"], IterableDataset):
+                # force the seed in the ``Dataset``
+                seed = int(os.getenv("PL_GLOBAL_SEED", 0)) + self.current_epoch + self.global_rank
```

Comment:
```suggestion
                seed = int(os.getenv("PL_GLOBAL_SEED", 0)) + self.global_rank
```

With current_epoch this might be not reproducible

Reply:
Without the `current_epoch`, the sampling would be the same if `reload_every_epochs` is True. Also, with progress tracking, we would have access to `current_epch`, so I believe it could be reproducible.

Reply:
The implementation we have for `pl_worker_init_function` function does not take the `current_epoch` too so I guess suffers from the same problem. Maybe this piece could reuse that function if we change it to return the seed where `worker_id` is not used if `None`

Even if we always have the current_epoch set, the user would need to modify it before running this code to mimic an older seed.

Not sure what would be the best solution here.